saleor/discount/utils.py

```diff
@@ -840,10 +842,10 @@ def _handle_order_promotion(
             ]
         )
 
-    _delete_gift_line(lines_info)
+    delete_gift_line(lines_info)
 
 
-def _delete_gift_line(lines_info: Iterable["CheckoutLineInfo"]):
+def delete_gift_line(lines_info: Iterable["CheckoutLineInfo"]):
     gift_line_infos = [line for line in lines_info if line.line.is_gift]
     line_ids_to_delete = [line_info.line.id for line_info in gift_line_infos]
     if line_ids_to_delete:
```

Comment:
btw. Can't we just make `CheckoutLine.objects.filter(checkout_id=line.checkout_id, is_gift=True).delete()`?

Reply:
I cannot, as in such a case the response from the mutation will contain this extra gift line